### DISCUSSION TITLE
Plugins - auth0-client - redis variables required

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#75] [Q&A] [answered]
Hi guys,
a question about the `auth0-client` plugin.

I know that the `auth0-client` handles users sessions through redis, is this mandatory?
Is there a way to use it just to interact with _Auth0_ without storing the user session on redis?

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, sessions needs to be stored somewhere and redis is used to do so, therefore it is required 😅 

Do you have a use case where you do not need anything stored? If so we could discuss an enhancement to the service.

**Reply by frengooo** [reply]
Why you need to store the session somewhere? If you want to use auth0 as custom `authentication-manager`, just to verify a JWT, you don't need to save it, do you?

Like the `client-credentials` behavior

**Reply by fredmaggiowski** [reply]
because you might use the session cookie which does not provide the JWT but a random UUID and instead caches the actual tokens; the only use case where redis may not be required is when you do not use session token and let auth0-client directly forward the tokens.. If you need such use case you can open a PR but I believe that the service should not start if you require some client to have the stored tokens

**Reply by frengooo** [reply]
We fixed by using another flow allowed by using `Envoy` and the JWT filter.

> I believe that the service should not start if you require some client to have the stored tokens

What do you mean? I didn't understand

**Reply by fredmaggiowski** [reply]
because if you try and use the website scope (for instance) and do not configure redis, the service won't be able to create new sessions and you'll only find that out at runtime when users try to login; this is something you want to catch as soon as possible instead.

**Reply by frengooo** [reply]
Ok clear thanks.
I don't know the details about the use case, I'll try to deeply investigate 😄
